components/src/core/Drawer/DrawerBody.tsx

```diff
@@ -12,64 +14,67 @@ const useStyles = makeStyles({
+    const classes = useStyles(bodyProps);
+    const { backgroundColor } = bodyProps;
+    const children = React.Children.toArray(bodyProps.children);
     return (
         <div className={classes.root} style={{ backgroundColor }}>
-            {children.map((child: any) => {
+            {children.map((child: any, index) => {
                 if (!child) {
                     return null;
                 }
 
-                const isNavGroup = child.type && child.type.displayName === 'DrawerNavGroup';
-                const groupProps = child.props;
-                return React.cloneElement(
-                    child,
-                    isNavGroup
-                        ? {
-                              activeBackgroundColor: groupProps.activeBackgroundColor || props.activeBackgroundColor,
-                              activeFontColor: groupProps.activeFontColor || props.activeFontColor,
-                              activeIconColor: groupProps.activeIconColor || props.activeIconColor,
-                              backgroundColor: groupProps.backgroundColor || props.backgroundColor,
-                              chevron: groupProps.chevron === undefined ? props.chevron : groupProps.chevron,
-                              fontColor: groupProps.fontColor || props.fontColor,
-                              iconColor: groupProps.iconColor || props.iconColor,
-                              onSelect: props.onSelect,
-                              open: props.open,
-                              titleColor: groupProps.titleColor || props.titleColor,
-                          }
-                        : {}
+                if (child.type && child.type.displayName !== 'DrawerNavGroup') return child;
+                const groupProps: DrawerNavGroupProps = child.props;
+
+                // for any DrawerNavGroup, if a prop is not set (undefined), inherit from the DrawerBody
+                // open and onItemSelect is always determined by DrawerBody
+                return (
+                    <DrawerNavGroup
+                        {...groupProps}
+                        key={index.toString()}
+                        activeItem={groupProps.activeItem || bodyProps.activeItem}
+                        activeItemBackgroundColor={
+                            groupProps.activeItemBackgroundColor || bodyProps.activeItemBackgroundColor
+                        }
+                        activeItemFontColor={groupProps.activeItemFontColor || bodyProps.activeItemFontColor}
+                        activeItemIconColor={groupProps.activeItemIconColor || bodyProps.activeItemIconColor}
+                        activeItemBackgroundShape={
+                            groupProps.activeItemBackgroundShape || bodyProps.activeItemBackgroundShape
+                        }
+                        backgroundColor={bodyProps.backgroundColor}
+                        chevron={groupProps.chevron === undefined ? bodyProps.chevron : groupProps.chevron}
```

Comment:
Why the mix of `||` and ternary operators? Would it make sense to functionalize this somehow to make it more readable/maintainable?

Reply:
the `a?b:c` is used only on boolean values and is necessary. Or are you thinking about turning this into a function?